components/DeviceList/deviceList.js

```diff
@@ -6,8 +6,8 @@ export default function DeviceList({ devices }) {
     <>
       <h1>Devices</h1>
       <ul>
-        {devices?.map((device, index) => (
-          <li key={index}>
+        {devices?.map((device, id) => (
+          <li key={id}>
```

Comment:
Haha, not quite! You're stil using the `index` except that you named it `id`!
What you want is to destructure `device` OR access the id on the object `device.id`.